docs/docs/dev/explore/graceful-join-leave.md

```diff
@@ -0,0 +1,113 @@
+# Graceful joining and leaving the mesh
+
+In order to avoid network interruption when new proxies are joining the mesh or the old proxies are leaving the mesh, we need to follow best practices.
+
+Every proxy in the mesh is represented by the `Dataplane` object. If an inbound is marked as unhealthy, this proxy is not included in the list of service destinations for clients.
+
+```yaml
+type: Dataplane
+mesh: default
+name: web-01
+networking:
+  address: 127.0.0.1
+  inbound:
+    - port: 11011
+      servicePort: 11012
+      health:
+        ready: false
+      tags:
+        kuma.io/service: backend
+```
+
+::: tip
+False flags are not serialized therefore you can see
+```yaml
+health:
+  ready: false
+```
+as
+```yaml
+health: {}
+```
+:::
+
+The lack of graceful behavior can be mitigated with [Retry](../policies/retry.md), [CircuitBreaker](../policies/circuit-breaker.md), and [HealthCheck](../policies/health-check.md)
+however it is recommended to properly configure the applications.
+
+## Joining
+
+To perform graceful joining, we need to first make sure the application is ready to serve traffic before it can be considered a valid traffic destination.
+
+:::: tabs :options="{ useUrlFragment: false }"
+
+::: tab "Kubernetes
+When Pod is converted to a Dataplane object it will be marked as unhealthy until Kubernetes considers all containers to be ready.
```

Comment:
Can we link to the k8s doc maybe?

Reply:
Which docs?

Reply:
About container lifecycle this one for e.g: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination